lib/broccoli/default-packager.js

```diff
@@ -0,0 +1,408 @@
+ * @method getVendorFiles
+ * @param {Object} files A list of files to include into `<file-name>.js`
+ * @param {Boolean} isMainVendorFile Boolean flag to indicate if we are dealing with `vendor.js` file
+ * @return {Object} An object with lists of files to be concatenated into `vendor.js` file.
+ */
+function getVendorFiles(files, isMainVendorFile) {
+  return {
+    headerFiles: files,
+    inputFiles: isMainVendorFile ? [`${ADDONS_FOLDER_NAME}/**/*.js`] : [],
+    footerFiles: isMainVendorFile ? [VENDOR_SUFFIX_FILE_PATH] : [],
+  };
+}
+
+/*
+ * Concatenates all application's javascript Broccoli trees into one, as follows:
+ * Given an input tree that looks like:
+ *
+ * ```
+ * addon-modules/
+ *   ember-ajax/
+ *   ember-data/
+ *   ember-engines/
+ *   ember-resolver/
+ *   ...
+ * bower_components/
+ *   usertiming/
+ *   sinonjs/
+ *   ...
+ * the-best-app-ever/
+ *   components/
+ *   config/
+ *   helpers/
+ *   routes/
+ *   ...
+ * vendor/
+ *   ...
+ *   babel-core/
+ *   ...
+ *   broccoli-concat/
+ *   ...
+ *   ember-cli-template-lint/
+ *   ...
+ * ```
+ *
+ * Produces a tree that looks like:
+ *
+ * ```
+ * assets/
+ *   the-best-app-ever.js
+ *   the-best-app-ever.map (if sourcemaps are enabled)
+ * ```
+ *
+ * @method packageApplicationJs
+ * @param {BroccoliTree} EmberApp broccoli tree
+ * @param {Object} options
+ * @return {BroccoliTree} Final Broccoli tree with application files.
+ */
+function packageApplicationJs(tree, options) {
+  return concat(tree, {
+    inputFiles: [`${options.name}/**/*.js`],
+    headerFiles: [
+      APP_PREFIX_FILE_PATH,
+    ],
+    footerFiles: [
+      APP_SUFFIX_FILE_PATH,
+      APP_CONFIG_FILE_PATH,
+      APP_BOOT_FILE_PATH,
+    ],
+    outputFile: options.outputFile,
+    annotation: 'Concat App',
+    sourceMapConfig: options.sourcemaps,
+  });
+}
+
+/*
+ * Concatenates all vendor's javascript Broccoli trees into one, as follows:
+ * Given an input tree that looks like:
+ *
+ * ```
+ * addon-modules/
+ *   ember-ajax/
+ *   ember-data/
+ *   ember-engines/
+ *   ember-resolver/
+ *   ...
+ * bower_components/
+ *   usertiming/
+ *   sinonjs/
+ *   ...
+ * the-best-app-ever/
+ *   components/
+ *   config/
+ *   helpers/
+ *   routes/
+ *   ...
+ * vendor/
+ *   ...
+ *   babel-core/
+ *   ...
+ *   broccoli-concat/
+ *   ...
+ *   ember-cli-template-lint/
+ *   ...
+ * ```
+ *
+ * Produces a tree that looks like:
+ *
+ * ```
+ * assets/
+ *   vendor.js
+ *   vendor.map (if sourcemaps are enabled)
+ * ```
+ *
+ * @method packageVendorJs
+ * @param {BroccoliTree} EmberApp broccoli tree
+ * @param {Object} options
+ * @return {BroccoliTree} Final Broccoli tree with vendor files.
+ */
+function packageVendorJs(tree, options) {
+  let importPaths = Object.keys(options.outputFilesMap);
+
+  let packagedVendorJsTrees = importPaths.map(importPath => {
+    let vendorObject = getVendorFiles(
+      options.outputFilesMap[importPath],
+      importPath === DEFAULT_CONFIG.outputPaths.vendor.js
```

Comment:
Please take into account this fix https://github.com/ember-cli/ember-cli/pull/7519